radix-engine/src/kernel/kernel_callback_api.rs

```diff
@@ -51,6 +51,7 @@ pub trait KernelCallbackObject: Sized {
     fn after_lock_substate<Y>(
         handle: LockHandle,
         size: usize,
+        first_time_lock: bool,
```

Comment:
Rename to something like `from_database`, as first time lock on heap nodes involves no database read?